api/v1alpha1/etcd_types.go

```diff
@@ -156,6 +165,9 @@ type EtcdSpec struct {
 	// VolumeClaimTemplate defines the volume claim template to be created
 	// +optional
 	VolumeClaimTemplate *string `json:"volumeClaimTemplate,omitempty"`
+	// VolumeRetentionPolicy defines the policy for retaining volumes on etcd scale-down and deletion
+	// +optional
```

Comment:
Not a strong opinion. I was thinking on the following lines. 
If we make it mutable,  it might result in ambiguous state representation. Like, If somebody scale down it initially without any retention. Then change the field value. It will see that even though retention policy is now to retain but the volumes are missing. Said that I agree to that counter argument that is result of user action we can't do anything about it. Hence no strong opinion.

But probably now you are thinking of "RetainAll" and "RetainMaster" (I am assuming master is leader in multi-node cluster), what if user hibernate when policy set to RetainMaster and the mutate field to RetainAll. The ambiguous state here is count of PVC is only 1 even though policy is RetailAll. 

So, user has to remember its history of action of CRD to know the reasoning for correct state of system in such scenarios.

Reply:
Also, regarding "VolumeClaimTemplate" its the name for volume. Statefulset will create PVC with this prefix. It this is changed, then there is possibility of orphan volumes. Again here PVC claim logic will delete those orphan volumes so its fine again. No strong opinion. So, fine let's keep it mutable.

Reply:
Agreed about the history part which can be solved by introducing the `volumeRetentionPolicy` in the status, I think.

Also, good point about `RetainMaster` meaning the leader. For better terminology, we should call it `RetainLeader`.